affinity_ocata/0001-allow-compute-nodes-to-be-associated-with-host-agg.patch

```diff
@@ -0,0 +1,247 @@
+From cc4070f78b764af439b429908a97f82e620803c9 Mon Sep 17 00:00:00 2001
+From: Roman Dobosz <roman.dobosz@intel.com>
+Date: Wed, 27 Dec 2017 13:51:25 +0100
+Subject: [PATCH] allow compute nodes to be associated with host agg
+
+This is basically an Ocata backport patch from Jay Pipes:
+https://review.openstack.org/#/c/526753
+---
+ nova/compute/api.py                                |  37 +++++-
+ nova/tests/functional/compute/__init__.py          |   0
+ .../tests/functional/compute/test_aggregate_api.py | 139 +++++++++++++++++++++
+ nova/tests/unit/compute/test_compute.py            |   4 +-
+ 4 files changed, 174 insertions(+), 6 deletions(-)
+ create mode 100644 nova/tests/functional/compute/__init__.py
+ create mode 100644 nova/tests/functional/compute/test_aggregate_api.py
+
+diff --git a/nova/compute/api.py b/nova/compute/api.py
+index 36de3abc34..b064f26a45 100644
+--- a/nova/compute/api.py
++++ b/nova/compute/api.py
+@@ -4540,6 +4540,32 @@ class AggregateAPI(base.Base):
+             availability_zones.update_host_availability_zone_cache(context,
+                                                                    host_name)
+ 
++    def _service_or_compute_node_exists(self, ctx, host_or_node):
++        """
++        Returns True if a service host or compute node record could be found
++        for the supplied host_or_node string. We first check to see if a
++        service record can be found with the host matching the host_or_node
++        parameter by looking at the host mapping records in the API database.
++        If we don't find a service record there, we then ask all cell databases
++        to find a compute node with a hypervisor_hostname matching the supplied
++        host_or_node parameter.
++        """
++        # NOTE(gryf): we don't handle cells in Ocata yet
++        try:
++            objects.Service.get_by_compute_host(ctx, host_or_node)
++            return True
++        except exception.HostMappingNotFound:
```

Comment:
Is this the right error? I thought the host mapping was a cells thing.

Reply:
Yeah, you're right, should be `exception.ComputeHostNotFound` instead. I'm scratching my head though, since I have it right on my codebase… Will send it fixed soon